source/simulator/core.cc

```diff
@@ -870,125 +870,157 @@ namespace aspect
   }
 
 
-
   template <int dim>
-  void
+  Table<2,DoFTools::Coupling>
   Simulator<dim>::
-  setup_system_matrix (const std::vector<IndexSet> &system_partitioning)
+  setup_system_matrix_coupling () const
   {
-    system_matrix.clear ();
+    Table<2,DoFTools::Coupling> coupling(introspection.n_components,
+                                         introspection.n_components);
 
-    bool have_fem_compositional_field = false;
-    for (unsigned int c=0; c<introspection.n_compositional_fields; ++c)
+    // Start by assuming nothing couples
+    coupling.fill (DoFTools::none);
+
+    const typename Introspection<dim>::ComponentIndices &x
+      = introspection.component_indices;
+
+    // The matrix-free solver does not work with melt transport
+    Assert(!(parameters.include_melt_transport && parameters.include_melt_transport), ExcNotImplemented());
```

Comment:
:-)

Reply:
Yes that was not quite the condition I had in mind :-) ... is fixed now.